tools/benchmark/src/mocha/customOutputRunner.ts

```diff
@@ -49,7 +49,14 @@ export function benchmarkCustom(options: CustomBenchmarkOptions): Test {
 		};
 
 		const startTime = timer.now();
-		await options.run(reporter);
+
+		try {
+			await options.run(reporter);
+		} catch (error) {
+			test.emit("fail", error);
+			throw error;
```

Comment:
I am also thinking about casting this to `throw error as Error;` as the unit test result below isn't returning `Test.err` of type `Error`.